src/components/CriteriaForm/CriteriaForm.tsx

```diff
@@ -14,27 +15,39 @@ import {
 } from '@taskany/bricks/harmony';
 
 import { GoalSelect } from '../GoalSelect/GoalSelect';
-import { GoalBadge } from '../GoalBadge';
+import { getStateProps, GoalBadge } from '../GoalBadge';
 import { FilterAutoCompleteInput } from '../FilterAutoCompleteInput/FilterAutoCompleteInput';
 import { AddInlineTrigger } from '../AddInlineTrigger/AddInlineTrigger';
 import { StateDot } from '../StateDot/StateDot';
+import { TaskBadge, TaskBadgeIcon } from '../TaskBadge/TaskBadge';
 
 import { tr } from './CriteriaForm.i18n';
 import s from './CriteriaForm.module.css';
 
+type GoalStateProps = ComponentProps<typeof StateDot>['state'];
+type TaskTypeProps = NonNullable<ComponentProps<typeof TaskBadge>['type']>;
+type TaskStateProps = NonNullable<ComponentProps<typeof TaskBadge>['state']>;
+
 interface SuggestItem {
     id: string;
     title: string;
-    state?: ComponentProps<typeof StateDot>['state'] | null;
+    state?: GoalStateProps | TaskStateProps | null;
+    type?: TaskTypeProps | null;
     _shortId: string;
 }
 
+const isGoalStateProps = (props: unknown): props is GoalStateProps =>
+    typeof props === 'object' && props != null && 'lightForeground' in props && 'darkForeground' in props;
+
+const isTaskStateProps = (props: unknown): props is TaskStateProps =>
+    typeof props === 'object' && props != null && 'src' in props && 'title' in props;
+
```

Comment:
May it be better to have some `type` props to separate different types of SuggestItem?